cwl/src/test/scala/wdl4s/cwl/CwlWorkflowWomSpec.scala

```diff
@@ -96,7 +97,52 @@ class CwlWorkflowWomSpec extends FlatSpec with Matchers {
     }
   }
 
-  "A WdlNamespace for 3step" should "provide conversion to WOM" in {
+  behavior of "A WdlNamespace for 3step"
+
+  it should "parse ECMAScript expressions for arguments" in {
+    val wf = decodeAllCwl(rootPath / "three_step.cwl").map {
+      _.select[Workflow].get
+    }.value.unsafeRunSync.fold(error => throw new RuntimeException(s"broken parse! msg was $error"), identity)
+
+    // The second step (aka 1) step should be cgrep
+    val run: WorkflowStep.Run = wf.steps.apply(1).run
+    val commandLineTool: CommandLineTool = run.select[CommandLineTool].get
+
+    commandLineTool.id.get should include("cgrep")
+
+    /**
+      * Runtime version of `refineMV[ECMAScriptExpression]("$(foo)")` as I cannot summon the dark magic for compile
+      * time checks.
+      *
+      * Compiler output after `import eu.timepit.refined._`:
+      *
+      *   Error:(_, _) could not find implicit value for parameter v: eu.timepit.refined.api.Validate[String,wdl4s.cwl.ECMAScriptExpression]
+      *     val foo = refineMV[ECMAScriptExpression]("$(foo)")
+      */
+    def toECMAScriptExpression(unrefined: String): ECMAScriptExpression = {
+      eu.timepit.refined.api.RefType.applyRef[ECMAScriptExpression](unrefined).right.get
+    }
+
+    import shapeless._
+    val tests = Table(
+      ("index", "result"),
+      (0, Inr(Inl("grep"))),
+      (1, Inl(toECMAScriptExpression("$(inputs.pattern)"))),
+      (2, Inl(toECMAScriptExpression("$(inputs.file)"))),
+      (3, Inr(Inl("|"))),
+      (4, Inr(Inl("wc"))),
+      (5, Inr(Inl("-l")))
+    )
+
+    forAll(tests) { (index, result) =>
+      val argument: CommandLineTool.Argument = commandLineTool.arguments.get.apply(index)
+      val commandLineBinding: CommandLineBinding = argument.select[CommandLineBinding].get
+      val stringOrExpression: CommandLineTool.StringOrExpression = commandLineBinding.valueFrom.get
+      stringOrExpression should be(result)
```

Comment:
ToL: any chance of making each case a `it should "correctly identify the expression X" in {}` case?